sycl/source/detail/plugin.hpp

```diff
@@ -175,7 +180,7 @@ class plugin {
     pi::emitFunctionEndTrace(CorrelationID, PIFnName);
     pi::emitFunctionWithArgsEndTrace(CorrelationIDWithArgs,
                                      static_cast<uint32_t>(PiApiOffset),
-                                     PIFnName, ArgsData.data(), R, *MPlugin);
+                                     PIFnName, ArgsDataPtr, R, *MPlugin);
```

Comment:
Should you also put lines 180-181 under if (xptiTraceEnabled())?

Reply:
I don't think so. Bodies of those functions are wrapped with `if (xptiTraceEnabled())` already. The main overhead here is from packing arguments.

Reply:
It just looks strange that the call to emitFunctionWithArgsBeginTrace is guarded above (I assume their bodies also have that check) but these are not. Anyways, I am fine as is if you don't want to change

Reply:
Right, I'm planning some more changes to shut down notifications, that are not received by any party. This is a temporary fix for the problem anyway.